core/src/main/java/com/nightscout/core/model/CookieMonsterDownload.java

```diff
@@ -0,0 +1,244 @@
+// Code generated by Wire protocol buffer compiler, do not edit.
+// Source file: /Users/klee/Projects/Nightscout/android-uploader/core/src/main/java/com/nightscout/core/protobuf/G4Download.proto
+package com.nightscout.core.model;
+
+import com.squareup.wire.Message;
+import com.squareup.wire.ProtoField;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.squareup.wire.Message.Datatype.ENUM;
+import static com.squareup.wire.Message.Datatype.STRING;
+import static com.squareup.wire.Message.Datatype.UINT32;
+import static com.squareup.wire.Message.Datatype.UINT64;
+import static com.squareup.wire.Message.Label.REPEATED;
+import static com.squareup.wire.Message.Label.REQUIRED;
+
+public final class CookieMonsterDownload extends Message {
```

Comment:
Protobuf handles changes to the model rather well. This is why most of the fields are marked as optional in the proposed definition. From the [protobuf docs](https://developers.google.com/protocol-buffers/docs/overview)

>You can add new fields to your message formats without breaking backwards-compatibility; old binaries simply ignore the new field when parsing. So if you have a communications protocol that uses protocol buffers as its data format, you can extend your protocol without having to worry about breaking existing code.